src/cpl.rs

```diff
@@ -324,30 +358,46 @@ impl FromStr for CslStringList {
     }
 }
 
-/// Convenience for creating a [`CslStringList`] from a slice of _key_/_value_ tuples.
-///
-/// # Example
-///
-/// ```rust, no_run
-/// use gdal::cpl::CslStringList;
-///
-/// let opts = CslStringList::try_from(&[("One", "1"), ("Two", "2"), ("Three", "3")]).expect("known valid");
-/// assert_eq!(opts.len(), 3);
-/// ```
-impl<const N: usize> TryFrom<&[(&str, &str); N]> for CslStringList {
-    type Error = GdalError;
-
-    fn try_from(pairs: &[(&str, &str); N]) -> Result<Self> {
+impl FromIterator<CslStringListEntry> for CslStringList {
+    fn from_iter<T: IntoIterator<Item = CslStringListEntry>>(iter: T) -> Self {
         let mut result = Self::default();
-        for (k, v) in pairs {
-            result.set_name_value(k, v)?;
+        for e in iter {
+            result.add_entry(&e).unwrap_or_default()
+        }
+        result
+    }
+}
+
+impl<'a> FromIterator<&'a str> for CslStringList {
+    fn from_iter<T: IntoIterator<Item = &'a str>>(iter: T) -> Self {
+        iter.into_iter()
+            .map(Into::<CslStringListEntry>::into)
+            .collect()
+    }
+}
+
+impl FromIterator<String> for CslStringList {
+    fn from_iter<T: IntoIterator<Item = String>>(iter: T) -> Self {
+        iter.into_iter()
+            .map(Into::<CslStringListEntry>::into)
+            .collect()
+    }
+}
+
+impl Extend<CslStringListEntry> for CslStringList {
```

Comment:
Too bad we can't use `CSLMerge` here.

Reply:
I actually started to implement it (and still can) but the [C++ implementation](https://github.com/OSGeo/gdal/blob/27e2cf443dea54b5fcb60fc2900480b7ef866448/port/cpl_string.cpp#L273) scared me as I saw no evidence that it would extend the length of the original array. That said, my C++ is 20 years rusty.

Reply:
That function just appends one element at a time.

Reply:
Couldn't find a `{re|m}alloc` in there or in `CSLSetNameValue` either.

Reply:
I think it ends up in https://github.com/OSGeo/gdal/blob/27e2cf443dea54b5fcb60fc2900480b7ef866448/port/cpl_string.cpp#L111.